Redesigned Player Setup

XSlayer300 opened the pull request:
<!-- ***STOP!***  Read this: If this is not a PR ready for review and merge or WIP, open it as a draft PR, using the arrow next to 'Create Pull Request'>

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Redesigns the player setup to this. I also added a brief lore on TGMC as well.

https://cdn.discordapp.com/attachments/498576438908420106/638017685946630194/unknown.png

Updated design as per player feedback:
https://cdn.discordapp.com/attachments/498576438908420106/638702659632758784/unknown.png

**Updated to include separate background pages:**
https://cdn.discordapp.com/attachments/498576438908420106/638730416638918656/unknown.png

The design is based on Baystation 12's, hence _some_ the layout is based on it.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Because redesigns are great and all, it also fleshes out too.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
tweak: Redesigned the player setup menu, based on Baystation 12.
add: Added a "Background" option for a headstart primer on TGMC.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Rohesie commented:
Damn good PR.
This could also use a xeno version, and game-mode specific ones, but it's a great start.

spookydonut commented:
uhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh ui clutter?

Tupinambis commented:
Wayyyyyyy too long. Keep it short and sweet, and keep the nice list instead of shoving things to the sides

Rohesie commented:
This kind of long is good. It detracts nothing from those who don't want lore, and adds flavor.

Rohesie commented on code/datums/gamemodes/crash.dm:
```diff
@@ -376,22 +376,22 @@
 
 	var/output = "<div align='center'>"
-	output += "<p><a href='byond://?src=[REF(NP)];lobby_choice=show_preferences'>Setup Character</A></p>"
+	output += "<i>You are part of the <b>TerraGov Marine Corps</b>, a military branch of the TerraGov council. Their mission is to enforce space law for the purpose of defending Earth's orbit as well as other solar colonies around the galaxy under the conflict of the Independent Colonial Confederation and intelligent Xenomorph aliens. \nThe TGMC is composed of willing men and woman all across TerraG ...
```
```suggestion
	output += "<i>You are part of the <b>TerraGov Marine Corps</b>, a military branch of the TerraGov council. Their mission is to enforce space law for the purpose of defending Earth's orbit as well as other solar colonies around the galaxy under the conflict of the Independent Colonial Confederation and the intelligent xenomorph threat. \nThe TGMC is composed by willing men and women from all kinds of social strata, hailing from all across the TerraGov systems. \nAs the vessel approaches to the ordered location on space, the cryostasis pods deactivate and awake you from your long-term stasis. Knowing that it's one of those days again, you hope that you'll make this out alive...</i>"
```

Rohesie commented on code/modules/mob/new_player/new_player.dm:
```diff
@@ -73,7 +74,7 @@
 	output += "</div>"
 
-	var/datum/browser/popup = new(src, "playersetup", "<div align='center'>New Player Options</div>", 240, 300)
+	var/datum/browser/popup = new(src, "playersetup", "<div align='center'>Welcome to TGMC - [SSmapping.configs[SHIP_MAP].map_name]</div>", 560, 325)
```
```suggestion
	var/datum/browser/popup = new(src, "playersetup", "<div align='center'>Welcome to TGMC[SSmapping?.configs ? "- [SSmapping.configs[SHIP_MAP].map_name]" : ""]</div>", 560, 325)
```
Test this. The current setup is runtiming because at the start `SSmapping` is not yet up.

Rohesie commented on code/datums/gamemodes/crash.dm:
```diff
@@ -376,22 +376,22 @@
 	output += "</div>"
 
-	var/datum/browser/popup = new(NP, "playersetup", "<div align='center'>New Player Options</div>", 240, 300)
+	var/datum/browser/popup = new(NP, "playersetup", "<div align='center'>Welcome to TGMC - [SSmapping.configs[SHIP_MAP].map_name]</div>", 560, 325)
```
```suggestion
	var/datum/browser/popup = new(NP, "playersetup", "<div align='center'>Welcome to TGMC[SSmapping?.configs ? "- [SSmapping.configs[SHIP_MAP].map_name]" : ""]</div>", 560, 325)
```

Rohesie commented on code/datums/gamemodes/distress.dm:
```diff
@@ -483,22 +483,22 @@
 	output += "</div>"
 
-	var/datum/browser/popup = new(NP, "playersetup", "<div align='center'>New Player Options</div>", 240, 300)
+	var/datum/browser/popup = new(NP, "playersetup", "<div align='center'>Welcome to TGMC - [SSmapping.configs[SHIP_MAP].map_name]</div>", 560, 325)
```
```suggestion
	var/datum/browser/popup = new(NP, "playersetup", "<div align='center'>Welcome to TGMC[SSmapping?.configs ? "- [SSmapping.configs[SHIP_MAP].map_name]" : ""]</div>", 560, 325)
```

Rohesie commented on code/datums/gamemodes/game_mode.dm:
```diff
@@ -713,20 +713,22 @@ Sensors indicate [numXenosShip ? "[numXenosShip]" : "no"] unknown lifeform signa
 	output += "</div>"
 
-	var/datum/browser/popup = new(NP, "playersetup", "<div align='center'>New Player Options</div>", 240, 300)
+	var/datum/browser/popup = new(NP, "playersetup", "<div align='center'>Welcome to TGMC - [SSmapping.configs[SHIP_MAP].map_name]</div>", 560, 325)
```
```suggestion
	var/datum/browser/popup = new(NP, "playersetup", "<div align='center'>Welcome to TGMC[SSmapping?.configs ? "- [SSmapping.configs[SHIP_MAP].map_name]" : ""]", 560, 325)
```

Rohesie commented on code/datums/gamemodes/distress.dm:
```diff
@@ -483,22 +483,22 @@
 
 	var/output = "<div align='center'>"
-	output += "<p><a href='byond://?src=[REF(NP)];lobby_choice=show_preferences'>Setup Character</A></p>"
+	output += "<i>You are part of the <b>TerraGov Marine Corps</b>, a military branch of the TerraGov council. Their mission is to enforce space law for the purpose of defending Earth's orbit as well as other solar colonies around the galaxy under the conflict of the Independent Colonial Confederation and intelligent Xenomorph aliens. \nThe TGMC is composed of willing men and woman all across TerraG ...
```
```suggestion
	output += "<i>You are part of the <b>TerraGov Marine Corps</b>, a military branch of the TerraGov council. Their mission is to enforce space law for the purpose of defending Earth's orbit as well as other solar colonies around the galaxy under the conflict of the Independent Colonial Confederation and intelligent Xenomorph aliens. \nThe TGMC is composed of willing men and women all across the TerraGov systems, from all races and quirks. \nAs the vessel approaches the ordered location on the map, the cryostasis pods deactivate and awake you from your long-term stasis. Knowing that it's one of those days again, you hope that you'll make this out alive...</i>"
```

Rohesie commented on code/datums/gamemodes/game_mode.dm:
```diff
@@ -713,20 +713,22 @@ Sensors indicate [numXenosShip ? "[numXenosShip]" : "no"] unknown lifeform signa
 
 	var/output = "<div align='center'>"
-	output += "<p><a href='byond://?src=[REF(NP)];lobby_choice=show_preferences'>Setup Character</A></p>"
+	output += "<i>You are part of the <b>TerraGov Marine Corps</b>, a military branch of the TerraGov council. Their mission is to enforce space law for the purpose of defending Earth's orbit as well as other solar colonies around the galaxy under the conflict of the Independent Colonial Confederation and intelligent Xenomorph aliens. \nThe TGMC is composed of willing men and woman all across TerraG ...
```
```suggestion
	output += "<i>You are part of the <b>TerraGov Marine Corps</b>, a military branch of the TerraGov council. Their mission is to enforce space law for the purpose of defending Earth's orbit as well as other solar colonies around the galaxy under the conflict of the Independent Colonial Confederation and intelligent Xenomorph aliens. \nThe TGMC is composed of willing men and women all across TerraGov systems, from all races and quirks. \nAs the vessel approaches the ordered location on the map, the cryostasis pods deactivate and awake you from your long-term stasis. Knowing that it's one of those days again, you hope that you'll make this out alive...</i>"
```

Rohesie commented on code/modules/mob/new_player/new_player.dm:
```diff
@@ -42,16 +42,17 @@
 
 	var/output = "<div align='center'>"
-	output += "<p><a href='byond://?src=[REF(src)];lobby_choice=show_preferences'>Setup Character</A></p>"
+	output += "<i>You are part of the <b>TerraGov Marine Corps</b>, a military branch of the TerraGov council. Their mission is to enforce space law for the purpose of defending Earth's orbit as well as other solar colonies around the galaxy under the conflict of the Independent Colonial Confederation and intelligent Xenomorph aliens. \nThe TGMC is composed of willing men and woman all across TerraG ...
```
```suggestion
	output += "<i>You are part of the <b>TerraGov Marine Corps</b>, a military branch of the TerraGov council. Their mission is to enforce space law for the purpose of defending Earth's orbit as well as other solar colonies around the galaxy under the conflict of the Independent Colonial Confederation and the intelligent xenomorph threat. \nThe TGMC is composed of willing men and women all across TerraGov systems, from all races and quirks. \nAs the vessel approaches to the ordered location on the map, the cryostasis pods deactivate and awake you from your long-term stasis. Knowing that it's one of those days again, you hope that you'll make this out alive...</i>"
```

Max-023 commented:
Change Earth to Terra and we good.

factoryman942 commented:
the window is too short to see the polls button without scrolling
also, eww wall of text; why not just have it spew into chat when you join as a marine or something?

Spyroshark commented:
you need to make the window height larger the polls button doesn't fit

spookydonut commented:
> 
> 
> the window is too short to see the polls button without scrolling
> also, eww wall of text; why not just have it spew into chat when you join as a marine or something?

agreed this is ui clutter that makes this ui harder to use

XSlayer300 commented:
Updated the design to be less... clutter. I added a separate background button for this.

https://cdn.discordapp.com/attachments/498576438908420106/638702659632758784/unknown.png

Rohesie commented on code/modules/mob/new_player/new_player.dm:
```diff
@@ -379,6 +382,17 @@
+	var/output = "<div align='center'>"
+	output += "<i>The <b>TerraGov Marine Corps'</b> mission is to enforce space law for the purpose of defending Terra's orbit as well as other solar colonies around the galaxy under the conflict of the Independent Colonial Confederation and the intelligent xenomorph threat. \nThe TGMC is composed by willing men and women from all kinds of social strata, hailing from all across the TerraGov systems. ...
+
+	output += "<i>Hailing from one of Saturn's colonies, as well as Mars and other unlisted habitats, the <b>xenomorph threat</b> remains at large. Extremely dangerous extraterrestrial lifeforms, part of the hive under the Queen Mother, had caught the TGMC and NT colonies off-guard during their discovery in 2414. \nThey are divided into castes, each with their specialized roles equivalent to a tradi ...
```
Maybe add this part next to "Join as a Xeno" and the other next to "Join the TGMC" ? Differentiating the backgrounds/stories?

XSlayer300 replied:
Like make it side-by-side?

XSlayer300 replied:
Tried a different solution this time, as per Spyro's request.

XSlayer300 commented:
Updated, this time with a Background button with selectable lore pages.

https://cdn.discordapp.com/attachments/498576438908420106/638730416638918656/unknown.png

factoryman942 commented:
![image](https://user-images.githubusercontent.com/20996272/67806657-3bab7000-fa8b-11e9-8434-0533da90ab15.png)

*scream
also why has this been testmerged for like two days with a "Do not TM" label on it?

factoryman942 commented:
![image](https://user-images.githubusercontent.com/20996272/67983367-71cc2980-fc1c-11e9-80f9-99827896a7ed.png)
it's still fucked
